internal/provisioner/controller/gatewayclass.go

```diff
@@ -181,6 +181,19 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 					invalidParamsMessages = append(invalidParamsMessages, msg)
 				}
 			}
+
+			if params.Spec.Envoy.ExtraVolumeMounts != nil {
```

Comment:
Please add a test case that covers this new code branch, similar to https://github.com/projectcontour/contour/blob/main/internal/provisioner/controller/gatewayclass_test.go#L243-L277